core/src/main/java/io/questdb/cairo/ColumnVersionWriter.java

```diff
@@ -102,6 +112,23 @@ public void truncate() {
         }
     }
 
+    public void upsert(long timestamp, ColumnVersionReader detachedColumnVersionReader) {
+        LongList detachedEntries = detachedColumnVersionReader.getCachedList();
+        LongList entries = getCachedList();
+        for (int i = 0, limit = detachedEntries.size(); i < limit; i += 4) {
+            // detachedEntries: [partitionTimestamp, columnIndex, columnNameTxn, columnTop]*
+            long columnTop = detachedEntries.getQuick(i + 3);
+            if (timestamp == detachedEntries.getQuick(i) && columnTop != entries.getQuick(i + 3)) {
```

Comment:
Why do we check `columnTop != entries.getQuick(i + 3)` ?
`entries` can have very different Indies for the same partition

Reply:
indeed